### ISSUE TITLE
createStreamableUI causes multiple rerenders / UI flickering

### ISSUE THREAD

**Issue by DennisKo** [vercel/ai#2939]
### Description

Use `createStreamableUI` and then `.append` a few items. Every `append` will cause the previous item to render/mount (I think?) again which causes weird UI flickering. It's even more obvious when you have some fade-in animation on the items.

https://github.com/user-attachments/assets/e56b5d9f-0300-4ad6-ad08-0e48eb66be96

### Code example

export async function getMovies(messages: Message[]) {
  'use server';
  const movieUi = createStreamableUI(null);

setTimeout(() => {
    movieUi.update(
      <Movie .../>,
    );
  }, 1000);

  setTimeout(() => {
    movieUi.append(
      <Movie .../>,
    );
  }, 2000);

  setTimeout(() => {
    movieUi.append(
      <Movie .../>,
    );
    movieUi.done();
  }, 3000);

  return movieUi.value;
}

### Additional context

`"ai": "3.3.29"`
`"next": "14.2.8"`

**Comment by lgrammel** [comment]
Potential duplicate of https://github.com/vercel/ai/issues/1850

**Comment by lgrammel** [comment]
Can you try using `createStreamableValue` instead and relying on client components as a workaround? I'll plan to take a look at the issue, but it might be related to a long-standing issue that requires next js changes.

**Comment by DennisKo** [comment]
> Can you try using createStreamableValue instead and relying on client components as a workaround? I'll plan to take a look at the issue, but it might be related to a long-standing issue that requires next js changes.

Would be quite a big refactor and kind of would defeat the purpose of streaming RSCs. (my app is basically a demo of that). 

What I noticed it also creates this nested `Suspense` tree. I don't think that's need for a 4 item array basically?

![suspense-tree](https://github.com/user-attachments/assets/ecea6c7b-4937-4eb9-8ae8-814a4a5b9bd0)

**Comment by lgrammel** [comment]
Yeah - so in a nutshell, the suspense tree is the original problem. #1825 fixes that. However, #1825 caused #1933 and needed to be reverted, and from my understanding #1933 is a next js issue. I plan to spend time investigating it but this is not trivial and may not be solvable in the AI SDK itself.

**Comment by james-pratama** [comment]
I think its also causing this bug in my app to cause it to flicker up after each message finishes streaming. Happened only after I upgraded to  @ai-sdk/openai": "^0.0.58" and "ai": "^3.3.32". Do you know what might be the most recent safe version without this bug? 

https://github.com/user-attachments/assets/ef5bbbbe-5965-4bb8-8055-62b82ca871c2

**Comment by james-pratama** [comment]
In case it might be helpful for anyone else, the most stable release I could find without the flicker where we also get OpenAI tokens usage data is 3.2.5

**Comment by lgrammel** [comment]
@james-pratama 3.2.5 is before the revert of #1933 - did you run into any other issues? what next.js and react versions are you using?

**Comment by james-pratama** [comment]
I am on React 18.2.0 and still on Next 14.2.2. Have not seen any issues so far but will report back if I do.

**Comment by leo-paz** [comment]
is there any progress here? wondering if react 19/next 15 fixes anything?

**Comment by aayush-kapoor** [comment]
Marking this as closed for now - if you face any regression or have a similar problem after migrating to the latest AI SDK version, please feel free to open a new issue (with possible reproduction)

**Comment by lgrammel** [comment]
## Identify issue type

Identified as **Bug**  
Confidence: **High**

### Rationale
The issue reports concrete incorrect behavior in the existing `createStreamableUI` API—previous content flickers or remounts on append—rather than requesting a new capability. Maintainer comments corroborate it as a known implementation problem, even if the underlying fix may depend on Next.js.

### Evidence
- Each `createStreamableUI.append` reportedly remounts or rerenders previous items, causing visible UI flickering and repeated fade-in animations.
- The reporter observed a deeply nested `Suspense` tree produced by appending items.
- A maintainer identified the suspense tree as an existing long-standing problem, referenced a reverted fix, and suggested `createStreamableValue` only as a workaround.
- Another user reported similar flickering after upgrading and found an older SDK release without it.
